src/test/unit/algorithms/ConnectionsTest.cpp

```diff
@@ -677,4 +677,39 @@ TEST(ConnectionsTest, testSaveLoad) {
   ASSERT_EQ(c1, c2);
 }
 
+TEST(ConnectionsTest, testCreateSegmentOverflow) {
+    const size_t LIMIT = std::numeric_limits<Segment>::max();
+    if(LIMIT <= 256) { //connections::Segment is too large (likely uint32), so this test would run, but memory 
```

Comment:
these are the new tests that check overflow in createSynapse/Segment methods; 
but I cannot override at runtime the connections::Segment type (this is where teplates, instead of typedefs, would be useful), so the test is effectively skipped, unless the user manually changes the Segment/Synapse to unsigned char.